src/internet_identity/src/vc_mvp.rs

```diff
@@ -0,0 +1,247 @@
+use crate::assets::CertifiedAssets;
+use crate::delegation::check_frontend_length;
+use crate::{delegation, hash, state, update_root_hash, LABEL_SIG, MINUTE_NS};
+use candid::Principal;
+use canister_sig_util::CanisterSigPublicKey;
+use ic_cdk::api::{data_certificate, time};
+use ic_cdk::trap;
+use ic_certified_map::{Hash, HashTree};
+use identity_core::common::Url;
+use identity_core::convert::FromJson;
+use identity_credential::credential::{Credential, CredentialBuilder, Subject};
+
+use canister_sig_util::signature_map::SignatureMap;
+use internet_identity_interface::internet_identity::types::vc_mvp::{
+    GetIdAliasResponse, IdAliasCredentials, PreparedIdAlias, SignedIdAlias,
+};
+use internet_identity_interface::internet_identity::types::{
+    AnchorNumber, FrontendHostname, IdentityNumber,
+};
+use serde::Serialize;
+use serde_bytes::ByteBuf;
+use serde_json::json;
+use vc_util::{
+    did_for_principal, vc_jwt_to_jws, vc_signing_input, vc_signing_input_hash, AliasTuple,
+    II_CREDENTIAL_URL_PREFIX, II_ISSUER_URL,
+};
+
+// The expiration used for signatures
+const SIGNATURE_EXPIRATION_PERIOD_NS: u64 = 10 * MINUTE_NS;
+
+pub struct InvolvedDapps {
+    pub(crate) relying_party: FrontendHostname,
+    pub(crate) issuer: FrontendHostname,
+}
+
+pub async fn prepare_id_alias(
+    identity_number: IdentityNumber,
+    dapps: InvolvedDapps,
+) -> PreparedIdAlias {
+    state::ensure_salt_set().await;
+    check_frontend_length(&dapps.relying_party);
+    check_frontend_length(&dapps.issuer);
+
+    let id_alias_principal = get_id_alias_principal(identity_number, &dapps);
```

Comment:
This feels a little backwards. I would have expected:
1. create `let pk = canister_sig_pk_der(...)`
2. `let id_alias_principal = Principal::self_authenticating(pk)`

Right now you are creating the `seed` twice, no?